ocaml/xapi/quicktest_cbt.ml

```diff
@@ -0,0 +1,101 @@
+(*
+ * Copyright (C) 2017 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Client
+open Quicktest_common
+let start session_id =
+  let open Client in
+
+  (* Helper for test failure due to unexpected error *)
+  let report_failure error test_name test =
+    failed test (Printf.sprintf "%s failed: %s" test_name
+                   (ExnHelper.string_of_exn error)) in
+
+
+  let test_assert ~test op ~msg =
+    try assert op with assert_failure -> failed test msg in
+
+  (* Overall test suite runs smaller unit tests *)
+  let cbt_test = make_test "Testing changed block tracking" 2 in
+  try
+    start cbt_test;
+
+    (* generate list of SRs attached to a host *)
+    let list_of_attached_srs =
+      let is_attached sr =
+        let pbds = SR.get_PBDs ~session_id ~rpc:!rpc ~self:sr in
+        List.fold_left (||) false
+          (List.map (fun pbd ->
+               PBD.get_currently_attached ~session_id ~rpc:!rpc ~self:pbd
+             ) pbds) in
+      List.filter is_attached (SR.get_all ~session_id ~rpc:!rpc) in
+
+    (* find lvm SR (that passes data_destroy) to attach new VDI *)
+    let sR = List.filter
```

Comment:
Hmm, I think we shouldn't need it, as the allowed_operations field of the SR should include `vdi_data_destroy` if it's supported. But we're not testing `data_destroy` here, so I'd say let's remove this block, as the test should work without it.